### PR TITLE
Move to using interface for component, satisfies

### PR DESCRIPTION
**Pull request by jcscottiii**
This patch is the beginning of enabling multiple versions by
abstracting the logic to a common interface between versions

### CODE REVIEW

File: models/components/versions/base/component.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	GetReferences() *common.GeneralReferences
+	GetVersion() float32
+}
+
+type Satisfies interface {
+	GetStandardKey() string
+	GetControlKey() string
+	GetNarrative() string
+	GetCoveredBy() common.CoveredByList
+}
```
Comment (afeld) [inline models/components/versions/base/component.go]: This probably warrants some developer-centric documentation (or at least a comment at the top of the file saying "we are using `interface`s to provide an abstraction between schema versions. See `some/path/` for the implementations of the different versions."

File: models/components.go [hunk 2]
```diff
@@ -21,14 +23,44 @@ type Components struct {
...
+func (c Component) GetVerifications() *common.VerificationReferences {
+	return &c.Verifications
+}
+
+func (c Component) GetReferences() *common.GeneralReferences {
+	return &c.References
+}
+
+func (c Component) GetAllSatisfies() []base.Satisfies {
+	// Have to do manual conversion
```
Comment (afeld) [inline models/components.go]: Minor: I'd add "...to the more generic `interface` `type`."

Comment (jcscottiii) [inline models/components.go]: Will do

**Comment by afeld** [comment]
Looks good overall! Mind adding tests to cover the lines that Codecov is complaining about?

File: models/components/versions/base/component.go (hunk 1 above)
Comment (jcscottiii) [inline models/components/versions/base/component.go]: done

**Pull request merged**